src/lib.rs

```diff
@@ -0,0 +1,371 @@
+    pub fn new(evg_project_location: &Path) -> Result<Self> {
+        let discovery_service = Arc::new(ResmokeProxy::new());
+        let multiversion_service = Arc::new(MultiversionServiceImpl::new(discovery_service)?);
+        let evg_config_service = Arc::new(EvgProjectConfig::new(evg_project_location)?);
+        let evg_config_utils = Arc::new(EvgConfigUtilsImpl::new());
+        let gen_fuzzer_service = Arc::new(GenFuzzerServiceImpl::new(multiversion_service));
+        let gen_task_service = Arc::new(GenerateTasksServiceImpl::new(
+            evg_config_service,
+            evg_config_utils,
+            gen_fuzzer_service,
+        ));
+
+        Ok(Self { gen_task_service })
+    }
+}
+
+/// A container for configuration generated for a build variant.
+#[derive(Debug, Clone)]
+struct GeneratedConfig {
+    /// References to generated tasks that should be included.
+    pub gen_task_specs: Vec<TaskRef>,
+    /// Display tasks that should be created.
+    pub display_tasks: Vec<DisplayTask>,
+}
+
+impl GeneratedConfig {
+    /// Create an empty instance of generated configuration.
+    pub fn new() -> Self {
+        Self {
+            gen_task_specs: vec![],
+            display_tasks: vec![],
+        }
+    }
+}
+
+/// Create 'generate.tasks' configuration for all generated tasks in the provided evergreen
+/// project configuration.
+///
+/// # Arguments
+///
+/// * `deps` - Dependencies needed to perform generation.
+/// * `config_location` - Pointer to S3 location that configuration will be uploaded to.
+pub fn generate_configuration(deps: Dependencies, config_location: &str) -> Result<()> {
+    let generate_tasks_service = deps.gen_task_service;
+
+    // We are going to do 2 passes through the project build variants. In this first pass, we
+    // are actually going to create all the generated tasks that we discover.
+    let generated_tasks = generate_tasks_service.build_generated_tasks(config_location)?;
+
+    // Now that we have generated all the tasks we want to make another pass through all the
+    // build variants and add references to the generated tasks that each build variant includes.
+    let generated_build_variants =
+        generate_tasks_service.generate_build_variants(generated_tasks.clone());
+
+    let generated_tasks = generated_tasks.lock().unwrap();
+    let task_defs: Vec<EvgTask> = generated_tasks
+        .values()
+        .flat_map(|g| g.sub_tasks())
+        .collect();
+
+    let gen_evg_project = EvgProject {
+        buildvariants: generated_build_variants.to_vec(),
+        tasks: task_defs,
+        ..Default::default()
+    };
+
+    std::fs::create_dir_all(CONFIG_DIR).unwrap();
+    let mut config_file = Path::new(CONFIG_DIR).to_path_buf();
+    config_file.push("evergreen_config.json".to_string());
+    std::fs::write(config_file, serde_json::to_string_pretty(&gen_evg_project)?)?;
+    println!("{}", serde_yaml::to_string(&gen_evg_project)?);
```

Comment:
For now, I'm both writing the generated configuration to a file and printing it out to the stdout. This is mostly for debugging purpose while the tool is being built out.